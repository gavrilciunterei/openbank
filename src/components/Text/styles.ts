import styled from 'styled-components';
import { StyeProps } from './types';

export const TextStyled = styled.p<StyeProps>`
  font-weight: ${(props) => (props.fontWeight ? props.fontWeight : 'lighter')};
  font-size: ${(props) => props.size};
  line-height: 1.2;
  max-width: ${(props) => props.maxWidth};
  color: ${(props) => props.color};
  margin-top: 0;
`;
