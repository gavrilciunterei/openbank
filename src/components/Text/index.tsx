import React from 'react';
import { TextStyled } from './styles';
import { TextProps } from './types';

function Text({ children, color, fontWeight, size, maxWidth }: TextProps) {
  return (
    <TextStyled
      color={color}
      fontWeight={fontWeight}
      size={size}
      maxWidth={maxWidth}
    >
      {children}
    </TextStyled>
  );
}

export default Text;
