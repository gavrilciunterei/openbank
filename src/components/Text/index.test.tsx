import '@testing-library/jest-dom/extend-expect';
import { render, screen } from '@testing-library/react';
import Text from './index';

test('Text test', () => {
  render(<Text children={'Prueba de texto'} />);
  screen.getByText('Prueba de texto');
});
