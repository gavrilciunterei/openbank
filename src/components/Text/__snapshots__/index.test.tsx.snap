// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Text 1`] = `
<p
  className="sc-bdvvtL jHGvXH"
  color="red"
>
  Hola
</p>
`;
